lib/expect.js

```diff
@@ -127,6 +139,42 @@ function overwriteWithCondition(utils, condition) {
   }
 }
 
+function overwriteAlwaysUseSuper(utils) {
+  const {flag} = utils;
+
+  return function(_super) {
+    return async function () {
+      const obj = this._obj;
+      const isControllerPromise = obj instanceof ControllerPromise;
+      if (!isControllerPromise) {
+        return _super.apply(this, arguments);
+      }
+      const {waitForValue} = obj;
+      if (!waitForValue) {
+        const result = await obj;
+        flag(this, 'object', result);
+        return _super.apply(this, arguments);
+      }
+
+      const resultPromise = waitForValue((obj) => {
+        try {
+          flag(this, 'object', obj);
+          _super.apply(this, arguments);
+          return true;
+        } catch(e) {
+          lastExpectError = e;
```

Comment:
What might cause `_super.apply` to throw? Is the return value `true` or `false` significant to anywhere downstream in the Promise chain?

Reply:
Added some comments to explain what the code is doing.